openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,799 @@
-                    tmpPI = e.data[pdTempPrimaryIndicator[i].toLowerCase()];
-
-                    if (tmpPI != undefined) {
-                        tmpPI = tmpPI.split(',');
-                        // Check and see what category the tmpPI belongs to
-                        var elementsParent = checkElementCategory(elementName);
-
-                        // Keep track of parents and their respective children
-                        for (var k in parentChildKey) {
-                            if(k == elementsParent) {
-                                var ep = elementsParent;
-                                var pos;
-                                pos = parentChildKey[ep].indexOf(elementName);
-                                if (!~pos) {
-                                    parentChildKey[ep].push(elementName);
-                                }
-                            }
+        // iterate over the layerAttributes to access the data
+        for (var o = 0; o < la.length; o++) {
+            var tempSum = 0;
+            var region = la[o].properties[selectedRegion];
+            var theme = name;
+            // check the operator type and compute accordingly
+            if (operator == "Average (ignore weights)") {
+                for (var p in la[o].properties) {
+                    // iterate over the indicator child keys
+                    for (var r = 0; r < tempIndicatorChildrenKeys.length; r++) {
+                        if (p == tempIndicatorChildrenKeys[r]) {
+                            // Sum the theme indicators
+                            tempSum = tempSum + la[o].properties[p];
                         }
-                        tmp[elementName] = parseFloat(tmpPI[m]);
-                        tmp.municipality = municipality[m];
-                        primaryIndicator[m] = tmp;
                     }
                 }
-            }
-
-            // Create the primary indicators obj continued 
-            for (var j = 0; j < munic_num.length; j++) {
-                tmp.municipality = municipality[j];
-                primaryIndicator[j] = tmp;
-            }
-
-            // Keep the primaryIndicator obj as is and make a session copy that 
-            // is to be modifyed by the project definition weights 
-            weightedPrimaryIndicator = JSON.parse( JSON.stringify( primaryIndicator ) );
-            sessionPrimaryIndicator = JSON.parse( JSON.stringify( primaryIndicator ) );
-
-            // Multiply the copy of primary indicator data by the weighted value
-            for(var p1 in weightedPrimaryIndicator) {
-                for(var p2 in weightedPrimaryIndicator[p1]) {
-                    if(tempWeight.hasOwnProperty(p2)) {
-                        weightedPrimaryIndicator[p1][p2] *= tempWeight[p2];
+                // Grab the average
+                var average = tempSum / tempIndicatorChildrenKeys.length;
+                indicatorString.push(region + '|'+ theme +'|'+ average);
```

Comment:
(This comment was automatically hidden by github after you pushed some changes on near code)
If you had an array such as indicatorInfo = [],
then you can do something like
indicatorData.push({"region": region, "theme": theme, "sum": sum})
And afterwards you can access it easily without parsing a string.